runtime/compiler/trj9/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -60,6 +151,14 @@ bool J9::RecognizedCallTransformer::isInlineable(TR::TreeTop* treetop)
       case TR::java_lang_Math_max_L:
       case TR::java_lang_Math_min_L:
          return TR::Compiler->target.cpu.isX86() && !comp()->getOption(TR_DisableMaxMinOptimization);
+
+      case TR::java_lang_String_toLowerHWOptimized:
+      case TR::java_lang_String_toLowerHWOptimizedDecompressed:
+      case TR::java_lang_String_toLowerHWOptimizedCompressed:
+      case TR::java_lang_String_toUpperHWOptimized:
+      case TR::java_lang_String_toUpperHWOptimizedDecompressed:
+      case TR::java_lang_String_toUpperHWOptimizedCompressed:
+         return comp()->cg()->getSupportsInlineStringCaseConversion() && TR::Compiler->target.cpu.isX86();
```

Comment:
Why do we need a `isX86()` check in a platform agnostic optimization? Shouldn't `getSupportsInlineStringCaseConversion()` cover us already as all code generators should be setting this flag if they want this inrinsic inlining?

Reply:
I suppose Z would not currently support this sort of transformation via Recognized Call Transformer as we pattern match the call at direct evaluator time. Ok we can leave the x86 check in place for now. I'll open up a task on my end to start using this new infrastructure and get rid of the platfrom specific `isX86()` check.